ua/node_id_test.go

```diff
@@ -391,3 +392,13 @@ func TestSetNamespace(t *testing.T) {
 		})
 	}
 }
+
+func compareErrors(err1, err2 error) bool {
```

Comment:
I think `reflect.DeepEqual` does exactly what `compareErrors` does. 

https://play.golang.org/p/iBDNJELixui

Reply:
The underlying stack trace captures will be different:

https://play.golang.org/p/1WqB7u4BUf7

Reply:
Damn. That's not nice since the error itself doesn't change. It doesn't really matter where it occurs.

Reply:
I've added a comment to this function referring to your play.golang.org link. My guess is that this will bite us at some point when users expect `reflect.DeepEqual` to work.

Reply:
Just got bitten by this in two ways. First, that `errors` now adds a prefix and then that the comparison did not work. I'll expose the compare function so that we can use that in tests but it would be nice if we could find a more generic solution.